src/results/resultHooks.js

```diff
@@ -1,31 +1,29 @@
 import { useCallback, useContext } from "react";
 import { searchContext } from "../search/searchContext";
 
-const BASE_URL = 'https://api.publicapis.org/entries';
+const BASE_URL = "https://api.publicapis.org/entries";
 
 export function useQueryResults() {
   const endpoint = useEndPoint();
-  return useCallback(
-    async (https) => {
-      const results = await fetch(endpoint());
-      const { entries } = await results.json();
-      return entries;
-    },
-    [endpoint]
-  );
+  return useCallback(async () => {
+    const results = await fetch(endpoint());
+    const { entries } = await results.json();
+    return entries;
+  }, [endpoint]);
 }
 
 function useEndPoint() {
-  const { category, HTTPSsupport } = useContext(searchContext);
+  const { category, HTTPSsupport, searchBarValue } = useContext(searchContext);
+  const url = searchBarValue ? `${BASE_URL}?title=${searchBarValue}&` : BASE_URL+'?';
   return useCallback(() => {
```

Comment:
Instead of these complex conditions, investigate how you can manage URL parameters simply using https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams